site/static/js/external-purchase-button.component.js

```diff
@@ -0,0 +1,76 @@
+import { utils } from 's72';
+import { AppComponent, bindAllComponents, render, h, attrs, getComponentElement } from 's72.ui';
+
+export default class ExternalPurchaseButton extends AppComponent {
+  constructor(props, context) {
+    super(props, context);
+    this.state = { loaded: false };
+  }
+
+  componentDidMount() {
+    this.updateState();
+    this.app.messagebus.subscribe('user-signed-in', () => this.updateState());
+    this.app.messagebus.subscribe('user-signed-out', () => this.updateState());
+    this.app.messagebus.subscribe('shopping-session-completed', session =>
+      this.shoppingSessionCompleted(session)
+    );
+  }
+
+  componentWillUnmount() {
+    this.app.messagebus.unsubscribe('shopping-session-completed', () => this.updateState());
+  }
+
+  updateState() {
+    if (this.props.slug && this.app.shoppingService.isValidType(this.props.slug)) {
+      return this.loadAvailability(this.props.slug)
+        .then(() => {
+          return this.setState({ loaded: true });
+        }, utils.noop)
+        .then(() => {
+          let e = getComponentElement(this);
+          e.classList.toggle('s72-show', this.shouldBeVisible);
```

Comment:
Is this necessary if the render function renders nothin when `!this.shouldBeVisible`?  I get uncomfortable when I see classes directly attached to component elements because more than once, we've had mystery space appearing around empty components due to thoughtless css changes.

Reply:
I hate it, I just copied it from the other button and figured it might have been done like that for a reason, like not leaving weird empty elements littering the DOM.

Reply:
Weird empty elements littering the DOM is better than weird empty elements with CSS attached littering the DOM!

But... meh.